scripts/main.js

```diff
@@ -0,0 +1,338 @@
+
+        const alink = document.createElement('a');
+        alink.setAttribute('rel', 'noopener noreferrer');
+        alink.setAttribute('href', '#');
+        alink.setAttribute('aria-label', '');
+        alink.textContent = tag;
+        litag.appendChild(alink);
+      });
+    }
+
+    if (!cardObj.desktop || cardObj.id === 'special') {
+      const cardButton = document.createElement('a');
+      cardButton.classList.add('interaction');
+      cardButton.setAttribute('rel', 'noopener noreferrer');
+      cardButton.setAttribute('href', `#${cardObj.id}`);
+      cardButton.setAttribute('aria-label', 'See Project');
+      cardButton.textContent = 'See Project';
+      cardButton.addEventListener('click', () => {
+        popupWindow(cardObj);
+      });
+
+      if (cardObj.id === 'card-1') {
+        info.appendChild(cardButton);
+      } else {
+        card.appendChild(cardButton);
+      }
+    }
+
+    gridContainer.appendChild(card);
+  });
+}
+
+renderCards();
+
+const form = document.querySelector('form');
+
+const errorMessage = document.querySelector('.error');
+
+const email = document.querySelector('#email');
+const nameInput = document.querySelector('#name');
+const messageInput = document.querySelector('#message');
+
+function showError(msg) {
+  errorMessage.textContent = msg;
+  errorMessage.classList.add('active');
+  email.classList.add('error-icon');
+}
+
+function checkemail() {
+  const emailValue = email.value.trim();
+
+  if (emailValue === '') {
+    showError('Email should not be blank');
+    return false;
+  }
+
+  const emailRegex = /[A-Z]/g;
+
+  if (emailValue.match(emailRegex)) {
+    showError(`Email field doesn't allow capital letters. It should be ${emailValue.toLowerCase()}`);
+    return false;
+  }
+
+  return true;
+}
+
+let myFormData = { inputName: '', email: '', message: '' };
+
+function setFormData() {
+  myFormData = JSON.parse(localStorage.getItem('formData'));
+
+  email.value = myFormData.email;
+  nameInput.value = myFormData.inputName;
+  messageInput.value = myFormData.message;
+}
+
+function populateStorage() {
+  localStorage.setItem('formData', JSON.stringify(myFormData));
+  setFormData();
+}
+
+if (!localStorage.getItem('formData')) {
+  populateStorage();
+} else {
+  setFormData();
+}
+
+nameInput.addEventListener('input', () => {
+  myFormData.inputName = nameInput.value;
+  populateStorage();
+});
+
+messageInput.addEventListener('input', () => {
+  myFormData.message = messageInput.value;
+  populateStorage();
+});
+
+email.addEventListener('input', () => {
+  errorMessage.textContent = '';
+  errorMessage.classList.remove('active');
+  email.classList.remove('error-icon');
+
+  myFormData.email = email.value;
+
+  populateStorage();
+});
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+
+  if (checkemail()) {
+    form.submit();
+    myFormData = { inputName: '', email: '', message: '' };
+    populateStorage();
+  }
+});
```

Comment:
- [ ] [OPTIONAL] Good you added these scripts. But the following lines of codes and files (main.js) will be required later on in the portfolio application development.
Implementing them now in this specific task is not a good idea, please try not to go beyond the task requirements.